pkg/go/graph/weighted_graph_builder_test.go

```diff
@@ -484,3 +484,104 @@ func TestValidGraphModel(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, 3, graph.nodes["job#can_read"].weights["user"])
 }
+
+func TestValidConditionalGraphModel(t *testing.T) {
+	t.Parallel()
+	model := `
+	model
+  		schema 1.1
+		type user
+        type role
+            relations
+                define assignee: [user]
+        type permission
+            relations
+                define assignee: [role#assignee, role#assignee with condX]
+				define member: [user, permission#member, permission#member with condX]
+        type job
+            relations
+                define can_read: assignee from permission
+				define can_view: [user] or can_view from owner
+				define owner: [job, job with condX]
+                define permission: [permission, permission with condX]
+		condition condX (x:int) {
+					x > 0
+				}
+	`
+	authorizationModel := language.MustTransformDSLToProto(model)
+	wgb := NewWeightedAuthorizationModelGraphBuilder()
+	graph, err := wgb.Build(authorizationModel)
+	require.NoError(t, err)
+	require.Len(t, graph.nodes, 12)
+	require.Len(t, graph.edges, 8)
+	edges, _ := graph.GetEdgesFromNode(graph.nodes["permission#assignee"])
+	require.Len(t, edges, 1)
+	conditions := edges[0].conditions
+	require.Empty(t, edges[0].tuplesetRelation)
+	require.Len(t, conditions, 2)
```

Comment:
should we assert on the conditions itself?  (i.e., check the individual array item)